New article: Merge Conflicts on Flexible Environments

sofietoft opened the pull request:
## Description

Added a new article about how to resolve merge conflicts with flexible environments.

Also fixed some environment refs in other articles while waiting for deployments to complete.

mclausen commented on umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md:
```diff
@@ -0,0 +1,72 @@
+
+* The log file containing a list of the conflicting files.
+* The clone link for the flexible environment.
+* The clone link for the mainline environment that flexible environment is connected to.
```
Conflicts can happen between any environment, that is merging together. 

Would it make sense to call them source and destination environment?

sofietoft commented:
@eshanrnh - I've updated the list under "Preparations".
I've added source and target in paranthesis - can you give it a look, and tell me if it makes sense to do? 🤔

eshanrnh commented on umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md:
```diff
@@ -0,0 +1,72 @@
+Before you start resolving the merge conflict, ensure you have the following things ready:
+
+* The log file containing a list of the conflicting files.
+* The clone link for the flexible (source) environment.
+* The clone link for the mainline (target) environment that flexible environment is connected to.
```
Are these interchanged? Isn't the mainline the source of the changes, and the flexible the target—since you're pulling from the mainline into flexible?

Another suggestion for clarity and consistency since the article mentions pushing/pulling terms: 

* The clone link for the mainline environment (where you're pulling changes from).
* The clone link for the flexible environment (where you're pulling changes into).

sofietoft replied:
Hm, good question!
The merge conflicts happens when you want to merge your changes from the Flex environment into the Mainline environment.
In this case, the Flex is the source and the mainline is the target environment.
You're pulling from the target first, in order to make sure it's in sync with the source environment, before deploying your changes to the target environment.

Does that make sense?
That's also how we're doing it throughout the Cloud docs, but usually there the "pulling" isn't mentioned, as this happens automatically when pushing from on environment to another 🤔

eshanrnh commented on umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md:
```diff
@@ -0,0 +1,72 @@
+---
+description: Merge conflicts can happen on a flexible environment when the same schema has been changed in both the source and target environment.
```
```suggestion
description: Merge conflicts can happen when the same schema has been modified in both the flexible environment and the mainline environment it's connected to.
```

eshanrnh commented on umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md:
```diff
@@ -0,0 +1,72 @@
+
+# Merge Conflicts on Flexible Environments
+
+A merge conflict happens when one file or item contains changes in more than one instance that's part of a deployment.
```
```suggestion
A merge conflict happens when a file or item contains conflicting changes in more than one environment during deployment.
```

eshanrnh commented on umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md:
```diff
@@ -0,0 +1,72 @@
+
+1. Download the log file provided from the failed pull (see image above).
+2. Clone the flexible environment [to your local machine](../../set-up/working-locally.md).
+3. Use the mainline environment clone link to add a new `git remote:
```
```suggestion
3. Use the mainline environment clone link to add a new `git remote`:
```

eshanrnh commented:
I've added my thoughts, @sofietoft 🙌
